dbt/adapters/default/impl.py

```diff
@@ -773,3 +773,45 @@ def convert_agate_type(cls, agate_table, col_idx):
         for agate_cls, func in conversions:
             if isinstance(agate_type, agate_cls):
                 return func(agate_table, col_idx)
+
+    ###
+    # Operations involving the manifest
+    ###
+    @classmethod
+    def run_operation(cls, profile, project_cfg, manifest, operation_name,
+                      result_key):
+        """Look the operation identified by operation_name up in the manifest
+        and run it.
+
+        Return an an AttrDict with three attributes: 'table', 'data', and
+            'status'. 'table' is an agate.Table.
+        """
+        operation = manifest.find_operation_by_name(operation_name, None)
+
+        # This causes a reference cycle, as dbt.context.runtime.generate()
+        # ends up calling get_adapter, so the import has to be here.
+        import dbt.context.runtime
+        context = dbt.context.runtime.generate(
+            operation,
+            project_cfg,
+            manifest.to_flat_graph(),
+        )
+
+        # TODO: should I get the return value here in case future operations
+        # want to return some string? Jinja (I think) stringifies the results
+        # so it's not super useful. Status, I guess?
+        operation.generator(context)()
+
+        # This is a lot of magic, have to know the magic name is 'catalog'.
+        # TODO: How can we make this part of the data set? Could we make it
+        # the operation's name/unique ID somehow instead?
+        result = context['load_result'](result_key)
+        return result
+
+    ###
+    # Abstract methods involving the flat graph
+    ###
+    @classmethod
+    def get_catalog(cls, profile, project_cfg, run_operation):
```

Comment:
there's some naming inconsistency here. the comment should read `manifest` instead of `flat graph`, `run_operation` should be `manifest` to be consistent with the underlying implementations